### PR TITLE
fix IE url bugs

### PR DESCRIPTION
**Pull request by ascott1**
This should fix the bug where URL hashes weren't being appended correctly. To be totally honest I don't love the way I did this, so I'm open to feedback and criticsm @KimberlyMunoz.

I would also appreciate a thorough manual testing before merging. 

Also, I'm not sure if this would effect multiple page search results and I can't test locally.

### CODE REVIEW

**Comment by ascott1** [comment]
This might solve an issue we don't need to solve. Hold off on merging this until we can confirm what needs to be done!

File: regulations/static/regulations/js/source/views/main/child-view.js [hunk 1]
```diff
@@ -151,15 +151,19 @@ var ChildView = Backbone.View.extend({
             var url = this.url,
                 hashPosition;
 
+            console.log(options);
```
Comment (KimberlyMunoz) [inline regulations/static/regulations/js/source/views/main/child-view.js]: Extra console.log!

Comment (ascott1) [inline regulations/static/regulations/js/source/views/main/child-view.js]: OH NO

File: regulations/static/regulations/js/source/views/main/child-view.js [hunk 2]
```diff
@@ -151,15 +151,19 @@ var ChildView = Backbone.View.extend({
...
             if (options && typeof options.scrollToId !== 'undefined') {
                 url += '#' + options.scrollToId;
-            }
-            else {
+            } else {
                 hashPosition = (typeof Backbone.history.fragment === 'undefined') ? -1 : Backbone.history.fragment.indexOf('#');
                 //  Be sure not to lose any hash info
                 if (hashPosition !== -1) {
-                    url += Backbone.history.fragment.substr(hashPosition);
+                    url = url.slice(0, hashPosition) + '#' + options.id;
```
Comment (KimberlyMunoz) [inline regulations/static/regulations/js/source/views/main/child-view.js]: I know this is going to get looked at again, but I think we could remove the ternary operators in L160 to make lines 156-168 to be something like this:

```
if (options && typeof options.scrollToId !== 'undefined') {
                url += '#' + options.scrollToId;
            } else {
                // Don't lose the hash info.
                if (typeof Backbone.history.fragment !== 'undefined') {
                    hashPosition =  Backbone.history.fragment.indexOf('#');
                    url = url.slice(0, hashPosition) + '#' + options.id;
                } else if (options.type !== 'diff')  {
                    url += '#' + options.id;
                }
            }
```

Comment (ascott1) [inline regulations/static/regulations/js/source/views/main/child-view.js]: :+1: I am not a fan of ternary operators in general.

Comment (KimberlyMunoz) [inline regulations/static/regulations/js/source/views/main/child-view.js]: Since hashPosition gets used right after, I think this is easier to read.

Comment (cmc333333) [inline regulations/static/regulations/js/source/views/main/child-view.js]: If you go with that approach, maybe take it one step further and remove the nested if:

```javascript
if (options && typeof options.scrollToId !== 'undefined') {
    url += '#' + options.scrollToId;
// Don't lose the hash info.
} else if (typeof Backbone.history.fragment !== 'undefined') {
    hashPosition =  Backbone.history.fragment.indexOf('#');
    url = url.slice(0, hashPosition) + '#' + options.id;
} else if (options.type !== 'diff')  {
    url += '#' + options.id;
}
```

It's a bit worry-some, though, that there's no default.

Comment (ascott1) [inline regulations/static/regulations/js/source/views/main/child-view.js]: @cmc333333 very nice. To ease your worry a bit, the default is just `url` which already exists in memory. These ifs would only be if something needs appended to the url. 

This is actually likely to be closed as we haven't ever been appending that hash to the URL and I'm not sure that it solves the IE problem I was hoping it solves.

**Comment by ascott1** [comment]
I updated this to also fix the IE internal linking bug and include the refactoring suggestions made by @KimberlyMunoz and @cmc333333. 

Now when a user clicks an internal link in IE the browser window scrolls to the correct position. I've verified this in IE10.

**Comment by KimberlyMunoz** [comment]
I tested this on IE 10 and IE 11 this morning and it fixes that strange routing bug :raised_hands:, but introduces a new bug when comparing differences that can lead to long urls like this, `http://localhost:3000/diff/1005-6/2012-12121/2011-11111?from_version=2011-1111#1005-6/2012-12121/2011-11111?from_version=2011-11111`.

**Comment by ascott1** [comment]
@KimberlyMunoz my last commit should fix this. Do you mind taking a peak and merging for me once you've tested?

**Comment by KimberlyMunoz** [comment]
Sure. Going to take a look now.

**Comment by KimberlyMunoz** [comment]
Great. I just checked this across multiple Internet Explorers and it looks to be working well.

**Comment by ascott1** [comment]
:boom: :tada: :dancers:

**Pull request merged**